pkg/gen/input/workflows/internal/file/create_release_branch.go

```diff
@@ -0,0 +1,136 @@
+      name: Check out the repository
+      with:
+        fetch-depth: 0  # Clone the whole history, not just the most recent commit.
+
+    - name: Fetch all tags and branches
+      run: "git fetch --all"
+
+    - name: Set up python dependencies
+      run: "pip3 install gitpython==3.1.3 semver==2.10.2"
+
+    - name: Create the script containing the branch logic
+      run: |
+        cat > ./create-branch.py <<-EOF
+        # This script creates a branch for the previous minor version based on the current commit's parent
+        # if the current tag introduces a new minor version and the release branch doesn't already exist.
+
+        import os
+        import semver
+        from git import Repo
+
+        p = "refs/tags/"  # GitHub tags start with this prefix
+
+        # Takes a string and returns the string without the given prefix, if it is present
+        def removeprefix(string: str, prefix: str) -> str:
+            if string.startswith(prefix):
+                return string[len(prefix):]
+            return string
+
+        # Takes a string and returns the semver VersionInfo for that string, even if it includes a leading 'v'
+        def version(string: str) -> semver.VersionInfo:
+            string = removeprefix(string, 'v')
+            return semver.VersionInfo.parse(string)
+
+        repo = Repo(os.getcwd())
+
+        # Get the current tag from env and strip quotation marks
+        current_tag = os.environ.get("GITHUB_REF").strip('"')
+
+        # Remove GitHub ref path
+        current_tag = removeprefix(current_tag, p)
+        print("Current tag is " + current_tag)
+
+        # Get the tag of the "first" parent of the current commit including suffix, just for human reference
+        parent_tag = repo.git.describe('--tags', "{}".format(repo.commit().parents[0]))
+        print("Parent commit tag was {}".format(parent_tag))
+
+        # Get the closest tag to the parent commit (the tag version without the suffix)
+        parent_tag = repo.git.describe('--tags', '--abbrev=0', "{}".format(repo.commit().parents[0]))
+
+        # Get the semver for the parent tag
+        parent_version = version(parent_tag)
+        print("Parent base version was {}".format(parent_tag))
+
+        # Get the semver for the current tag
+        current_version = version(current_tag)
+
+        # Format the expected name of the release branch for the previous minor version
+        previous_branch_name = "release-v{}.{}.x".format(parent_version.major, parent_version.minor)
+
+        # Check if the current tag version introduces a new major or minor version
+        new_version = False
+        if current_version.major > parent_version.major:
+          print("Current tag is a new major version")
+          new_version = True
+        elif current_version.major == parent_version.major and current_version.minor > parent_version.minor:
+          print("Current tag is a new minor version")
+          new_version = True
```

Comment:
I'm not sure what's going on here but before I dive in I'd like to clear something.

For a given tag `v1.2.3` we want to make sure that branch `release-v1.2.x` exists correct?

If so why we can't just take the tag, transform its name into the desired branch name and create the branch if it doesn't exist?

Reply:
We agreed to create the branch only once a _new_ minor (or major) is tagged. So if we had `v1.2.3` already, the branch `release-1.2.x` would be created when `v1.3.0` is tagged (or `v2.0.0`). So since this action runs on the "new" tag, it has to look backwards at the "old" version to know what to name the branch